README.md

```diff
@@ -20,4 +20,302 @@ composer require bedita/php-sdk
+    $relatedData = [
+        [
+            'id' => 9999, // another doc id
+            'type' => 'documents',
+        ],
+    ];
+    $client->addRelated($document['data']['id'], 'documents', 'see_also', $relatedData); // arguments passed are int|string $id, string $type, string $relation, array $relationData
+```
+
+`replaceRelated(int|string $id, string $type, string $relation, array $data, ?array $headers = null)` is handy to replace relation data.
+
+```php
+    // replace related data, in this example a "see_also" relation between document 8888 and document 9999
+    $relatedData = [
+        [
+            'id' => 9999,
+            'type' => 'documents',
+        ],
+    ];
+    $client->replaceRelated(8888, 'documents', 'see_also', $relatedData); // arguments passed are int|string $id, string $type, string $relation, array $relationData
+```
+
+Note: `addRelated` uses `post`, `replaceRelated` uses `patch`. Both call `/:type/:id/relationships/:relation`
+
+### Delete and restore data
+
+#### Soft delete
+
+Soft delete puts object into the trashcan.
+You can trash an object with `delete(string $path, mixed $body = null, ?array $headers = null)` or `deleteObject(int|string $id, string $type)`.
+
+```php
+    // delete annotation by ID 99999
+    $response = $client->delete('/annotations/99999'); // argument passed is string $path
+
+    // delete annotation by ID 99999 and type documents
+    $response = $client->deleteObject(99999, 'annotations'); // arguments passed are string|int $id, string $type
+```
+
+#### Restore data
+
+Data in trashcan can be restored with `restoreObject(int|string $id, string $type)`.
+
+```php
+    // restore annotation 99999
+    $response = $client->restoreObject(99999, 'annotations'); // arguments passed are string|int $id, string $type
+```
+
+#### Hard delete
+
+Hard delete removes object from trashcan.
+You can remove an object from trashcan with `remove(int|string $id)`.
+
+```php
+    // delete annotation by ID 99999
+    $response = $client->deleteObject(99999, 'annotations'); // arguments passed are string|int $id, string $type
+
+    // permanently remove annotations 99999
+    $response = $client->remove(99999); // argument passed is string|int $id
+```
+
+#### Remove relation
+
+Relation data can be removed using `removeRelated(int|string $id, string $type, string $relation, array $data, ?array $headers = null)`.
+
+```php
+    // remove related data, in this example a "see_also" relation between document 8888 and document 9999
+    $relatedData = [
+        [
+            'id' => 9999,
+            'type' => 'documents',
+        ],
+    ];
+    $client->removeRelated(8888, 'documents', 'see_also', $relatedData); // arguments passed are int|string $id, string $type, string $relation, array $relationData
+```
+
+### Upload a file
+
+Use `upload(string $filename, string $filepath, ?array $headers = null)` to perform a `POST /streams/upload/:filename` and create a new stream with your file.
```

Comment:
I think we should deprecate this one.

The easiest, by far, way of create a new media object via file upload is simply 

```
        $headers = ['Content-type' => 'image/jpeg'];
        $fileTmp = file_get_contents($filePath);

        $response = $apiClient->post(
            sprintf('/images/upload/%s', $fileName),
            $fileTmp,
            $headers
        );
```

This will create the stream stream resource, the image object and link the two in a single call.